core/src/main/java/org/apache/accumulo/core/client/lexicoder/BigDecimalLexicoder.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.accumulo.core.client.lexicoder;
+
+import java.io.ByteArrayInputStream;
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.math.BigDecimal;
+import java.math.BigInteger;
+
+import org.apache.accumulo.core.clientImpl.lexicoder.FixedByteArrayOutputStream;
+
+public class BigDecimalLexicoder extends AbstractLexicoder<BigDecimal> {
```

Comment:
```suggestion
/**
 * A lexicoder to encode/decode a BigDecimal to/from bytes that maintain its native Java sort order.
 *
 * @since 2.1.0
 */
public class BigDecimalLexicoder extends AbstractLexicoder<BigDecimal> {
```